website/docs/reference/api-tokens-and-client-keys.mdx

```diff
@@ -18,25 +18,28 @@ This section describes what API tokens are. For information on how to create the
 
 :::
 
-Use API tokens to connect to the Unleash server API. API tokens come in three distinct types:
+Use API tokens to connect to the Unleash server API. API tokens come in four distinct types:
 
 -   [Admin tokens](#admin-tokens)
+-   [Personal access tokens](#personal-access-tokens)
 -   [Client tokens](#client-tokens)
 -   [Front-end tokens](#front-end-tokens)
 
 All types use [the same format](#format) but have different intended uses. Admin and client tokens are _secrets_ and should _not_ be exposed to end users. Front-end tokens, on the other hand, are not secret.
 
 ### The parts of an API token {#token-data}
 
-An API token contains the following pieces of information:
+Admin, client and front-end tokens contain the following pieces of information:
 
 | Name | Description |
 | --- | --- |
-| Token name (sometimes called "username") | The token's name. Names or **not** required to be unique. |
+| Token name (sometimes called "username") | The token's name. Names are **not** required to be unique. |
 | Type | What kind of token it is: admin, client, or front-end. |
 | Projects | What projects a token has access to. |
 | Environment | What environment the token has access to. |
 
+Personal access tokens follow their own special format, and only contain an optional description for the token and an expiry date.
```

Comment:
Thanks for this! There's a few things that are still a little unclear to me. I've also changed the description to a table for consistency. What do you think? And is it correct?


Also, when you create a PAT via the API, what endpoint is it? What sort of payload do you use?

```suggestion
Personal access tokens follow a different format and consist of the following pieces of information: 

| Name | Description |
| --- | --- |
| Description | A short description of what the token is to be used for. Must be unique. |
| Expiry data | The time when the token will stop working. |
```

Reply:
It doesn't actually need to be a unique description right now but I think that's a bug actually. I think your description of how this works what it should be - a unique description. I'm okay with taking your text as is and patching PATs to take on this behaviour before we release it

You can create a PAT via the API. I'm not really sure you should though, because it'd be a pain. You'd need to do one of two things, either get a a cookie via the login mechanism that the front end uses and use that in your request or get a PAT and use that PAT in your request to create a new PAT.

If you can get one of those two tokens, you can call {{base_url}}/api/admin/user/tokens with the following packet:

```
{
    "description": "Some stuff",
    "expiresAt": "2022-12-09T09:41:21.705Z"
}
```

Is this something we want to document and encourage?

Reply:
>  I'm okay with taking your text as is and patching PATs to take on this behaviour before we release it

Yeah, I'm down with that! I think making it unique is sensible if that is the only human-readable identifier we have. In that case, it's kind of like a "name"?

> I'm not really sure you should though, because it'd be a pain. 

Right, because a PAT can only be created for your own user. An admin can't create a PAT **for** someone, right?

The question was mostly to ensure that it's a separate endpoint, so the `type`, and `projects` etc. doesn't really come into play at all.

Reply:
Nah, you gotta be you to make a token for yourself. The admin user can make a token for the admin user but you can't make one on behalf of someone else (at least not not and I don't think we have plans to do so). It's very much a separate endpoint, for the same reasons you point out - it would be chaos to use the existing ones